apps/api/src/domain/factories/complexFactories/buildBibliographicReferenceFactory/index.ts

```diff
@@ -0,0 +1,25 @@
+import { DTO } from '../../../../types/DTO';
+import bibliographicReferenceValidator from '../../../domainModelValidators/bibliographicReferenceValidator';
+import { isValid } from '../../../domainModelValidators/Valid';
+import { BookBibliographicReference } from '../../../models/bibliographic-reference/entities/book-bibliographic-reference.entity';
+import { IBibliographicReference } from '../../../models/bibliographic-reference/interfaces/IBibliographicReference';
+import { InstanceFactory } from '../../getInstanceFactoryForEntity';
+import getCtorFromBibliographicReferenceType from './getCtorFromBibliographicReferenceType';
+
+const bibliographicReferenceFactory: InstanceFactory<IBibliographicReference> = (
+    input: unknown
+) => {
+    const validationResult = bibliographicReferenceValidator(input);
+
+    // Return error if the dto does not satisfy domain model invariants- CONSIDER THROWING
```

Comment:
Do some UX testing to experiment with this ID. Throwing (what is returned as a 500 anyway) would really clean up the code. It is technically a system error.

Reply:
But I now remember that we want to let the client decide how to handle the failure. It may be ok to throw in the `fetchById` case, but we usually want to filter in the case of the `fetchMany` behaviour. Alternatively, we can throw there too, but we would want good logging in production and to be notified if this actually happens.